File: src/solver/sat/DPLLResult.java
package solver.sat;

import java.util.Set;

public class DPLLResult {
    public SATInstance instance;
    public Model model;
    public boolean isSAT;

    public DPLLResult(SATInstance instance, Model model, boolean isSAT) {
        this.instance = instance;
        this.model = model;
        this.isSAT = isSAT;
    }

    public String createSolutionString(Set<Integer> vars) {
        StringBuffer buf = new StringBuffer();
        for (Integer var : vars) {
            if (this.model.model.contains(var)) {
                buf.append(var + " true ");
                if (this.model.model.contains(-var)) {
                    System.out.println("-var is there and var too!");
                }
            } else { // NOTE: assign false to arbitrary choice vars
                buf.append(var + " false ");
            }
        }
        String result = buf.toString();
        return result.substring(0, result.length() - 1);
    }
}

File: src/solver/sat/DPLLResult.cpp
#include "include/DPLLResult.hpp"
#include "include/utils.hpp"

DPLLResult::DPLLResult(SATInstance* instance, Model* model, bool isSat) {
    this->instance = instance;
    this->model = model;
    this->isSAT = isSat;
}

std::string DPLLResult::createSolutionString(std::unordered_set<int>* vars) {
    // generates the string of variable assignments that is mapped to by the
    // "Solution" key in the final result printed by the solver
    std::string result = "";
    for (int var : *vars) {
        if (setContains(this->model->model, var)) {
            result += std::to_string(var) + " true ";
        } else {
            result += std::to_string(var) + " false ";
        }
    }
    return result.substr(0, result.length() - 1);
}